### PR TITLE
feat: add new `eslint-config-node` for Node.js projects

### PR DESCRIPTION
**Pull request by emmenko**
This is a slightly simpler/different ESLint config for Node.js projects. We can use this in our backend services.

### CODE REVIEW

File: packages-backend/eslint-config-node/index.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    /**
+     * @type {import('@babel/core').TransformOptions}
+     */
+    babelOptions: {
+      presets: [['@babel/preset-env', { targets: { node: 'current' } }]],
+    },
+  },
+
+  env: {
+    browser: false,
```
Comment (tdeekens) [inline packages-backend/eslint-config-node/index.js:27]: 💯

File: packages-backend/eslint-config-node/index.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  parser: '@babel/eslint-parser',
+
+  parserOptions: {
+    sourceType: 'module',
+    requireConfigFile: false,
+    /**
+     * @type {import('@babel/core').TransformOptions}
+     */
+    babelOptions: {
+      presets: [['@babel/preset-env', { targets: { node: 'current' } }]],
```
Comment (tdeekens) [inline packages-backend/eslint-config-node/index.js:22]: Should we maybe allow passing this as an option to not force anybody using this config to use the current version of Node.js? 🤔

Comment (emmenko) [reply packages-backend/eslint-config-node/index.js:22]: We can make it configurable yes.

Comment (emmenko) [reply packages-backend/eslint-config-node/index.js:22]: Hmm actually I'm not sure how to make it configurable, it's not like Babel.

Btw, the `current` node version is the same as `process.versions.node`. This respects the active node version, like when using `.nvmrc` file.

image

I think it should be fine like this. Or do you have a more specific use case in mind?

Comment (tdeekens) [reply packages-backend/eslint-config-node/index.js:22]: Ah okay. I understood differently what `current` means. If it's `process.env.node` it should be fine. As long as people respect the nvmrc.

Comment (emmenko) [reply packages-backend/eslint-config-node/index.js:22]: Yes that's what it says in the Babel docs: https://babeljs.io/docs/en/options#targetsnode

image

Comment (tdeekens) [reply packages-backend/eslint-config-node/index.js:22]: Sounds good. I think in internal services I just wanted to be more careful when it's around building.

File: packages-backend/eslint-config-node/index.js [hunk 3]
```diff
@@ -0,0 +1,222 @@
+// This is a workaround for https://github.com/eslint/eslint/issues/3458
+require('@rushstack/eslint-patch/modern-module-resolution');
+
+const { statusCode, allSupportedExtensions } = require('./helpers/eslint');
+// const { craRules } = require('./helpers/rules-presets');
```
Comment (CarlosCortizasCT) [inline packages-backend/eslint-config-node/index.js:5]: Maybe this line is a leftover?

File: packages-backend/eslint-config-node/LICENSE [hunk 4]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2020 commercetools GmbH
```
Comment (tdeekens) [inline packages-backend/eslint-config-node/LICENSE:3]:
```suggestion
Copyright (c) 2022 commercetools GmbH
```

**Review by tdeekens — APPROVED** [review]
Looks good now. Let's start using it and adjust when needed.

**Review by CarlosCortizasCT — APPROVED** [review]
lgtm

File: packages-backend/eslint-config-node/LICENSE (hunk 4 above)
Comment (emmenko) [reply packages-backend/eslint-config-node/LICENSE:3]: Yeah I'll do this as a follow up.

PS: I'm wondering if we can automate this with a github action or something that runs periodically once a year.

**Pull request merged**